src/core/request.js

```diff
@@ -59,31 +59,31 @@ Request.prototype.request = function(servers, callback_) {
   this.on('error', function() {});
   this.emit('request', this.remote);
 
-  function doRequest() {
-    if (Array.isArray(servers)) {
-      servers.forEach(function(server) {
-        self.setServer(server);
-        self.remote.request(self);
-      }, self);
-    } else {
-      self.remote.request(self);
+  const timeout = setTimeout(() => {
+    if (typeof callback === 'function') {
+      callback(new RippleError('tejTimeout'));
```

Comment:
I was thinking of `error` events: "Error events are treated as a special case in Node.js. If there is no listener for it, then the default action is to print a stack trace and exit the program." So I guess it isn't a problem because it is a `timeout` event, not an `error` event.